design/open/pull-request.md

```diff
@@ -0,0 +1,28 @@
+# Pull-request Proposal
+
+Enable a way to submit changes to a branch by creating a request to merge changes.
+The pull-request captures the merge operation we request to apply.
+
+### Goals
+
+- Capture request to merge a branch or specific reference
+- Mechanism to review, discuss and approve changes before applying to a branch
```

Comment:
Just merges for now, right?
```suggestion
- Mechanism to review, discuss and approve merges before applying to a branch
```

Reply:
Not going to change the support with pre-merge - it will trigger when you merge the PR as it is equal to a merge operation.
The PR trigger 'pull-request' action after internal merge operation is done without conflicts.
The new event event will enable check the changes before approving the merge.